posts/navarroaxel/06.how-to-use-databases-in-the-Docker-world.md

```diff
@@ -0,0 +1,182 @@
+---
+title: How to use databases in the Docker world
+published: true
+description: Run DB engines locally on our machine without installing them by using Docker containers.
+tags: 'docker, database, mongodb, postgres'
+cover_image: ./assets/docker-databases-cover.jpg
+id: 1059901
+---
+
+We can run DB engines locally on our machine without installing them by using Docker containers.
+
+💡 Remember that the container's data is removed with the container itself, so, we're going to store the DB data into a specific [Docker volume](https://docs.docker.com/storage/volumes/) (`--volume`) to keep it in time.
+
+## MongoDB
+
+We can start a MongoDB instance with this simple command:
+
+```bash
+docker run --detach --name mongo \
+  --publish 27017:27017 \
+  --volume mongo:/data/db \
+  mongo
+```
+
+This starts a Docker container that runs the `mongod` process and listens the default MongoDB port `27017`. By default MongoDB uses the `test` database and only accepts connections from `localhost` without authentication.
+
+### The Docker run command
+
+Let's explain the `docker run` arguments:
+
+1. `--detach` runs the container in background, like a service.
+2. `--name` assigns a specific name to the created container.
+3. `--publish` forwards a specific port in `localhost` to the given container.
+4. `--volume` creates or reuse a volume and attach it to the given container in the specified path.
+
+By default MongoDB uses the `/data/db` directory to store the databases (remember to mount a Docker volume in that path).
+
+### A MongoDB root user
+
+You can specify a `root` user to the DB adding the following environment variables to your the container
+
+```bash
+docker run --detach --name mongo \
+  -env MONGO_INITDB_ROOT_USERNAME=mongoadmin \
+  --env MONGO_INITDB_ROOT_PASSWORD=secret \
+  --publish 27017:27017 \
+  --volume mongo:/data/db \
+  mongo
+```
+
+For more variables check the `mongo` image page in [Docker hub](https://hub.docker.com/_/mongo).
+
+### Using the MongoDB shell
+
+You can connect to your DB via terminal by using the `mongosh` command in the same container.
+
+```bash
+docker exec -it mongo mongosh
+```
+
+Use `--username` and `--password` if those values are required.
+
+### How to update the engine
+
+We can update a DB instance easily if we have previously stored the DB data in a Docker volume ‼️. This is important, otherwise we'll lose data.
+
+1. Pull the latest MongoDB image
+
+    ```bash
+    docker pull mongo
+    ```
+
+1. First stop and remove the Docker container
+
+    ```bash
+    docker stop mongo
+    docker rm mongo
+    ```
+
+1. Recreate the container with the same given arguments, remember to include the `--env` variables.
```

Comment:
Which env variables? There aren't any here, it seems.

Reply:
I explain 2 optional variables (e.g. `--env MONGO_INITDB_ROOT_PASSWORD=secret`) above, I'm making a reference to them.

Reply:
I believe the --env variables are clear since you've mentioned before. Anyway, you could also add them here with sth like this:
```suggestion
1. Recreate the container with the same given arguments, remember to include the `--env` variables (in this case, just MONGO_INITDB_ROOT_USERNAME and MONGO_INITDB_ROOT_PASSWORD).
```

Reply:
👍